x-pack/plugins/observability/public/pages/cases/helpers.ts

```diff
@@ -23,31 +105,35 @@ export const useFetchAlertDetail = (alertId: string): [boolean, TopAlert | null]
 
   useEffect(() => {
     const abortCtrl = new AbortController();
+
     const fetchData = async () => {
       try {
         setLoading(true);
-        const response = await http.get<Record<string, unknown>>('/internal/rac/alerts', {
+        const response = await http.get<Record<string, unknown>>(BASE_RAC_ALERTS_API_PATH, {
           query: {
             id: alertId,
           },
+          signal: abortCtrl.signal,
         });
         if (response) {
           const parsedAlert = parseAlert(observabilityRuleTypeRegistry)(response);
+
           setAlert(parsedAlert);
-          setLoading(false);
         }
       } catch (error) {
         setAlert(null);
+      } finally {
+        setLoading(false);
       }
     };
 
-    if (!isEmpty(alertId) && loading === false && alert === null) {
```

Comment:
I was running into rerender issues here as well when I added the abort signal, but removing these dependencies seemed to fix it. I don't think we want to rely on loading here because it'll change during the fetch.

Reply:
you still want to check that is not loading or you can end up loading the same data twice.